### PR TITLE
Implement sorting for files tab

### PR DESCRIPTION
**Pull request by abika**
Tested all sorting types * reverse + save/restore setting.

Screenshot:
screenshot_1491492719

-> #305

### CODE REVIEW

File: src/ch/blinkenlights/android/vanilla/FileSystemAdapter.java [hunk 1]
```diff
@@ -105,7 +128,24 @@ public int compare(File a, File b)
...
+					case SORT_TIME:
+						order = Long.valueOf(a.lastModified())
+								.compareTo(Long.valueOf(b.lastModified()));
+						break;
+					case SORT_EXT:
+						order = FileUtils.getFileExtension(a.getName())
+								.compareToIgnoreCase(FileUtils.getFileExtension(b.getName()));
+						break;
+					default:
+						order = a.getName().compareToIgnoreCase(b.getName());
```
Comment (adrian-bl) [inline src/ch/blinkenlights/android/vanilla/FileSystemAdapter.java]: Please add an explicit case for SORT_NAME and have 'default' throw an IllegalArgumentException()

File: src/ch/blinkenlights/android/vanilla/LibraryActivity.java [hunk 2]
```diff
@@ -759,7 +759,7 @@ public boolean onPrepareOptionsMenu(Menu menu)
 		LibraryAdapter adapter = mCurrentAdapter;
 		boolean isLibraryAdapter = (adapter != null && adapter.getMediaType() != MediaUtils.TYPE_FILE);
 		menu.findItem(MENU_GO_HOME).setVisible(!isLibraryAdapter);
```
Comment (adrian-bl) [inline src/ch/blinkenlights/android/vanilla/LibraryActivity.java]: Just change this to
```
setVisible(adapter != null && adapter.getMediaType() == MediaUtils.TYPE_FILE)
```

and get rid of isLibraryAdapter

File: src/ch/blinkenlights/android/vanilla/LibraryPagerAdapter.java [hunk 3]
```diff
@@ -471,6 +466,7 @@ public void clearLimiter(int type)
 				mPendingFileLimiter = null;
 			} else {
 				mFilesAdapter.setLimiter(null);
+				loadSortOrder(mFilesAdapter);
```
Comment (adrian-bl) [inline src/ch/blinkenlights/android/vanilla/LibraryPagerAdapter.java]: Why is this needed?

File: src/ch/blinkenlights/android/vanilla/LibraryPagerAdapter.java [hunk 4]
```diff
@@ -607,6 +603,7 @@ public int setLimiter(Limiter limiter)
 				// but if it doesn't we would end up at the same scrolling position in a new
 				// folder which is uncool.
 				mLists[MediaUtils.TYPE_FILE].setSelection(0);
+				loadSortOrder(mFilesAdapter);
```
Comment (adrian-bl) [inline src/ch/blinkenlights/android/vanilla/LibraryPagerAdapter.java]: See above comment

File: src/ch/blinkenlights/android/vanilla/MediaAdapter.java [hunk 5]
```diff
@@ -604,12 +557,9 @@ public int getDefaultSortMode()
 		return 0;
 	}
 
-	/**
-	 * Return the current sort mode set on this adapter.
-	 */
-	public int getSortMode()
-	{
-		return mSortMode;
+	@Override
```
Comment (adrian-bl) [inline src/ch/blinkenlights/android/vanilla/MediaAdapter.java]: Please add a boilerplate comment describing the function.

File: src/ch/blinkenlights/android/vanilla/SortableAdapter.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2017 Adrian Ulrich <adrian@blinkenlights.ch>
```
Comment (Adonai) [inline src/ch/blinkenlights/android/vanilla/SortableAdapter.java]: You should be the copyright holder of this piece :)

File: src/ch/blinkenlights/android/vanilla/FileSystemAdapter.java (hunk 1 above)
Comment (abika) [inline src/ch/blinkenlights/android/vanilla/FileSystemAdapter.java]: Done.

File: src/ch/blinkenlights/android/vanilla/LibraryActivity.java (hunk 2 above)
Comment (abika) [inline src/ch/blinkenlights/android/vanilla/LibraryActivity.java]: That is not the same logic but I guess you want that improved.
Done.

File: src/ch/blinkenlights/android/vanilla/LibraryPagerAdapter.java (hunk 3 above)
Comment (abika) [inline src/ch/blinkenlights/android/vanilla/LibraryPagerAdapter.java]: Ehhm....
I thought the files adapter must be updated in the same way as the media adapters. But actually I dunno why.
Removed.

File: src/ch/blinkenlights/android/vanilla/LibraryPagerAdapter.java (hunk 4 above)
Comment (abika) [inline src/ch/blinkenlights/android/vanilla/LibraryPagerAdapter.java]: Removed.

File: src/ch/blinkenlights/android/vanilla/MediaAdapter.java (hunk 5 above)
Comment (abika) [inline src/ch/blinkenlights/android/vanilla/MediaAdapter.java]: Done.

File: src/ch/blinkenlights/android/vanilla/SortableAdapter.java (hunk 6 above)
Comment (abika) [inline src/ch/blinkenlights/android/vanilla/SortableAdapter.java]: Most of the code is copy&pasted. And likely it will be modified in the future. Don't like this copyleft stuff and git blame is enough.

**Comment by abika** [comment]
Squash if you want.

File: src/ch/blinkenlights/android/vanilla/LibraryPagerAdapter.java (hunk 3 above)
Comment (adrian-bl) [inline src/ch/blinkenlights/android/vanilla/LibraryPagerAdapter.java]: Hmm.. looking at the existing code, i don't see a reason why loadSortOdrer() was called for any other adapter as the sort order never depends on any limiter settings and stays valid for the lifetime of the adapter.

I'll remove the loadSortOrder() calls for the other adapter types after the merge

**Comment by adrian-bl** [comment]
Thanks again for the PR: Looks all fine now :-)
